base/shell/explorer/trayntfy.cpp

```diff
@@ -33,6 +34,255 @@ typedef struct _SYS_PAGER_COPY_DATA
     NOTIFYICONDATA  nicon_data;
 } SYS_PAGER_COPY_DATA, *PSYS_PAGER_COPY_DATA;
 
+
+struct IconWatcherData
+{
+    HANDLE hProcess;
```

Comment:
Please note that by keeping open a handle to a process, we are keeping a reference to the process,
and thus on process exit the kernel object is kept around.
This might influence apitests, since we now have a process inside reactos that keeps handles open to all processes with a tray icon.

Reply:
An open process handle and a reference are 2 different things in the kernel, but I'm just nitpicking :)

Anyway, on process exit we're signaled and close our handle. If you notice any zombies after this change, then there's a bug in the code.

Reply:
As long as you are aware of this, i'm happy :)